src/main/java/org/bih/eos/converter/cdm_field/date/DVToDateConverter.java

```diff
@@ -102,16 +103,37 @@ private Optional<Date> getIntervalEventStart(IntervalEvent intervalEvent) {
 		}
 	}
 
+        /**
+	 * Parses a {@link TemporalAccessor} from an openEHR DvDateTime into a {@link Date}.
+	 * <p>
+	 * This method is designed to be robust against different date-time representations from openEHR. It handles two primary cases
+	 * by inspecting the provided {@link TemporalAccessor} before parsing:
+	 * <ul>
+	 *     <li><b>With Time Zone Offset:</b> If the input contains an offset (e.g., "2022-02-03T04:05:06+02:00"),
+	 *     it is parsed directly into an {@link OffsetDateTime} to preserve the exact moment in time.</li>
+	 *     <li><b>Without Time Zone Offset:</b> If the input lacks an offset (e.g., "2022-02-03T04:05:06"), it is treated as a
+	 *     {@link LocalDateTime}. To convert this to a specific instant (which {@link Date} represents), the system's default time zone is applied.</li>
+	 * </ul>
+	 * This avoids {@link DateTimeParseException} for valid local date-times and ensures consistent conversion.
+	 * @param dvDateTime The {@link TemporalAccessor} to parse, typically from a {@link DvDateTime}.
+	 * @return An {@link Optional} containing the parsed {@link Date}, or an empty Optional if parsing fails or the input is partial/null.
+	 */
 	public Optional<Date> parseDvDateTime(TemporalAccessor dvDateTime) {
 		if (dvDateTime == null || !dvDateTime.isSupported(ChronoField.DAY_OF_MONTH)) {
 			LOG.warn("Date was empty, or partial (e.g. Day missing), therefore Date is returned empty.");
 			return Optional.empty();
 		} 
 
 		try {
-			OffsetDateTime dateTime = OffsetDateTime.parse(dvDateTime.toString());
-			Instant asInstant = dateTime.toInstant();
-			return Optional.of(Date.from(asInstant));
+			if (dvDateTime.isSupported(ChronoField.OFFSET_SECONDS)) {

```

Comment:
@prabash please move the inside of the  try block to an own method to improve readabiltiy.